File: 33th/jongmlee/ExpressionOfNumber.java
import java.util.*;

class ExpressionOfNumber {
    public int solution(int n) {
        int answer = 0;
        int[] arr = new int[n + 1];
        for (int i = 1; i < arr.length; i++) {
            arr[i] = arr[i - 1] + i;
        }
        //System.out.println(Arrays.toString(arr));
        for (int i = arr.length - 1; i >= 0; i--) {
            for (int j = i - 1; j >= 0; j--) {
                if (arr[i] - arr[j] == n) {
                    answer++;
                } else if (arr[i] - arr[j] > n) {
                    break;
                }
            }
        }
        return answer;
    }
}

File: 33th/jeakim/expression_of_number.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
	int answer = 0;
	int st = 1;

	while (st <= n){
		int num = st;
		for (int end = st + 1; num <= n; end++)
		{
			if (num == n){
				answer++;
				cout << st << ' ' << end << endl;
				break;
			}
			num += end;
		}
		st++;
	}
    return answer;
}

int main()
{
	cout << solution(16) << endl;

	return (0);
}